### PR TITLE
Replace the eservice json database with a more functional database that can be used for all service types

### PR DESCRIPTION
**Pull request by cmickeyb**
Replace the eservice database with more general service database

The eservice database was limited by what it could store (just enclave services), how it was stored (full file load/save), and how the information was exported.
    
The old eservice database is replaced by a persistent database (implemented with lmdb) that stores information about all services. The database is loaded on first use (similar to the block store manager) from settings in the configuration file.

The eservice_db pdo-shell command is replaced with one for all service databases (using the new format).

### CODE REVIEW

**Review by bvavala — APPROVED** [review]
Nice generalization that takes advantage of lmdb (a dependency that pdo already has).
I don't have comments.

**Review by prakashngit — APPROVED** [review]
Looks good, leaving a few questions mostly for clarification. Thank you.

File: python/pdo/service_client/service_data/service_data.py [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+## =================================================================
+
+# -----------------------------------------------------------------
+# The service database contains five stores which include three for
+# the services (one each for eservice, pservice, and sservice), one
+# to map service identities (often the verifying key for the service)
+# to its URL, and one to map a human readable name to a service url.
+# -----------------------------------------------------------------
+class ServiceDatabaseManager(object) :
+    map_size = 1 << 40
```
[the remark this answers is not in the document]
Comment (cmickeyb) [reply python/pdo/service_client/service_data/service_data.py:286]: Space is allocated on demand (sparse file). So actual size is proportional amount of data in the db.

File: python/pdo/service_client/service_data/service_data.py [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+
+        service_db = self.service_db(service_type)
+        with self.service_db_env.begin(write=False) as txn :
+            cursor = txn.cursor(db=service_db)
+            for service_url, packed_service_data in cursor :
+                yield (service_url, self.unpack_service_info(service_url, packed_service_data, service_type))
+
+    # -----------------------------------------------------------------
+    def store_by_url(self, url, service_type='eservice', service_names=[], service_identity=None) :
+        if service_type not in self.service_types :
```
[the remark this answers is not in the document]
Comment (cmickeyb) [reply python/pdo/service_client/service_data/service_data.py:511]: so you can specify more than one at a time.

File: python/pdo/service_client/service_data/service_data.py [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+            raise RuntimeError("provided identity does not match service")
+
+        new_info = cls(url, client.enclave_id, service_names=service_names)
+
+        # although we are picking up the information from the
+        # eservice here, we still need to verify it with the
+        # ledger; this creates a bit of redundancy but if we
+        # assume the connection is cached it shouldn't be much
+        # of a performance burden
+        new_info.last_verified_time = 0
```
[the remark this answers is not in the document]
Comment (cmickeyb) [reply python/pdo/service_client/service_data/service_data.py:119]: 0 is a flag to indicate that it has not been verfied. that is... its in the past.

File: build/tests/service-test.sh [hunk 4]
```diff
@@ -94,22 +94,23 @@ done
...
-try pdo-eservicedb --loglevel ${F_LOGLEVEL} add -u http://${F_SERVICE_HOST}:7102 -n es7102
-try pdo-eservicedb --loglevel ${F_LOGLEVEL} add -u http://${F_SERVICE_HOST}:7103 -n es7103
-try pdo-eservicedb --loglevel ${F_LOGLEVEL} add -u http://${F_SERVICE_HOST}:7104 -n es7104
-try pdo-eservicedb --loglevel ${F_LOGLEVEL} add -u http://${F_SERVICE_HOST}:7105 -n es7105
+try pdo-service-db --loglevel ${F_LOGLEVEL} clear
+try pdo-service-db --loglevel ${F_LOGLEVEL} add --url http://${F_SERVICE_HOST}:7101 --name es7101 --type eservice
+try pdo-service-db --loglevel ${F_LOGLEVEL} add --url http://${F_SERVICE_HOST}:7102 --name es7102 --type eservice
+try pdo-service-db --loglevel ${F_LOGLEVEL} add --url http://${F_SERVICE_HOST}:7103 --name es7103 --type eservice
+try pdo-service-db --loglevel ${F_LOGLEVEL} add --url http://${F_SERVICE_HOST}:7104 --name es7104 --type eservice
+try pdo-service-db --loglevel ${F_LOGLEVEL} add --url http://${F_SERVICE_HOST}:7105 --name es7105 --type eservice
```
[the remark this answers is not in the document]
Comment (cmickeyb) [reply build/tests/service-test.sh:109]: this is just testing the bash interface. the pservice and sservice are tested fully in the service-storage test.

**Review by marcelamelara — APPROVED** [review]
This is a super useful update, thank you!

**Pull request merged**